### PR TITLE
Ftr: Add dubbo-go-cli telnet tool support

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
**What this PR does**:
增加了dubbo-go的cli命令行工具，可以方便用户直连特定服务，发送包含特定参数的rpc请求，并展现结果。

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
支持嵌套struct，目前只支持单个参数的请求包和回包。
数据类型由于需要在json中定义，只支持golang基本数据类型：string、整形、浮点。

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

File: tools/cli/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# dubbo-go-cli
+
+### 1. 解决问题
```
Comment (AlexStocks) [inline tools/cli/README.md:3]: hey, guy, pls using English in an Apache project. Or u rename this file to readme_cn.md and add an English readme.

File: tools/cli/client/client.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+				Version:     version,
+				Method:      method,
+				Group:       group,
+				Params:      []interface{}{reqPkg},
+			},
+		},
+	}, nil
+}
+
+func createTCPAddr(host string, port int) string {
```
Comment (AlexStocks) [inline tools/cli/client/client.go:85]: pls using net.JoinHostPort instead.

File: tools/cli/client/client.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	if err != nil {
+		log.Fatalln("error: handler.Writer err = ", err)
+	}
+	startTime := time.Now()
+
+	// init rsp Package and add to pending queue
+	seqId := t.addPendingResponse(userPkg)
+	defer t.removePendingResponse(seqId)
+
+	requestDataChannel := make(chan []byte)
```
Comment (AlexStocks) [inline tools/cli/client/client.go:145]: do not need to set its length for this chan?

File: tools/cli/client/client.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+	doneChannel := make(chan bool)
+	responseDataChannel := make(chan []byte)
```
Comment (AlexStocks) [inline tools/cli/client/client.go:147]: the same as above

File: tools/cli/common/protocol.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+package common
+
+import (
+	"github.com/apache/dubbo-go/tools/cli/protocol"
+)
+
+var (
+	protocols = make(map[string]func() protocol.Protocol)
```
Comment (AlexStocks) [inline tools/cli/common/protocol.go:25]: pls set its init length param, such as 8

File: tools/cli/example/README.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
+# dubbo-go-cli 使用示例
```
Comment (AlexStocks) [inline tools/cli/example/README.md:1]: the same as above readme. pls rename this file to readme_cn.md and add an english readme.

**Comment by LaurenceLiZhixin** [comment]
Now waiting for pr: https://github.com/apache/dubbo-go-hessian2/pull/243 to be merged. 
Then I'll refer the latest hessian2 pkg and complete this pr.

**Comment by LaurenceLiZhixin** [comment]
I fix the change of hessian pkg, and problems above. Now I think it's able to merge.
Reviewer can try example in Readme file, to check if it works as hoped.

File: tools/cli/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+`$ sh build.sh`\
+and you can get dubbo-go-cli 
+
+### 3. Quick start：[example](example/README_CN.md)
+
+
+
+### Third party dependence（temporary）
+
+github.com/LaurenceLiZhixin/dubbo-go-hessian2 \
```
Comment (AlexStocks) [inline tools/cli/README.md:20]: why using ur branch?

File: tools/cli/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+github.com/LaurenceLiZhixin/json-interface-parser
```
Comment (AlexStocks) [inline tools/cli/README.md:21]: pls submit ur code to github.com/dubbogo/gost.

**Comment by LaurenceLiZhixin** [comment]
I'm waiting for gost v1.9.10, so as to make it build it successfully. Now is 1.9.9 and is not supported.

**Comment by LaurenceLiZhixin** [comment]
This pr can be merged only if pr #879 merged.
So, justing waiting.

**Comment by AlexStocks** [comment]
have u finished ur work? if so, pls fix the github action failure.

**Comment by LaurenceLiZhixin** [comment]
I have finished my work.

> have u finished ur work? if so, pls fix the github action failure.

**Review by cityiron — APPROVED** [review]

File: tools/cli/client/client.go [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+	"go.uber.org/atomic"
+)
+
+import (
+	"github.com/apache/dubbo-go/tools/cli/common"
+	"github.com/apache/dubbo-go/tools/cli/protocol"
+	_ "github.com/apache/dubbo-go/tools/cli/protocol/dubbo"
+)
+
+const defaultBufferSize = 1024 * 1024 * 4
```
Comment (AlexStocks) [inline tools/cli/client/client.go:38]: add comment for this constant to let the reader know what is the meaning of it.

**Review by AlexStocks — APPROVED** [review]

**Review by watermelo — APPROVED** [review]

**Pull request merged**